tss-esapi/src/traits.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::Result;
+
+/// Trait for types that can be converted into
+/// TPM marshalled data.
+pub trait Marshall: Sized {
+    const BUFFER_SIZE: usize;
+    /// Returns the type in the form of marshalled data
+    fn marshall(&self) -> Result<Vec<u8>>;
+}
+
+/// Trait for types that can be created from
+/// TPM marshalled data.
+pub trait UnMarshall: Sized {
+    /// Creates the type from marshalled data.
+    fn un_marshall(marshalled_data: &[u8]) -> Result<Self>;
```

Comment:
I think this can just be `fn unmarshall`, similar to how the library functions are named

Reply:
What's the benefit over just an `impl TryFrom<&[u8]>`?

Reply:
fixed

Reply:
> What's the benefit over just an impl TryFrom<&[u8]>?

I'd say this is a clearer representation of the fact that you're (un)marshalling structures with some functions intended for that purpose. For some types of structures (`TPM2B_...`) it wouldn't be clear at all that the input/output is the marshalled version and not the contents. For others it could, maybe, be more easily understood, but I think it's nice to make it explicit and consistent.